libcaf_core/src/blocking_actor.cpp

```diff
@@ -396,12 +398,14 @@ void blocking_actor::receive_impl(receive_cond& rcc,
         }
       }
     } while (skipped && !timed_out);
-    if (timed_out)
+    bool post_condition = rcc.post();
+    if (timed_out) {
       bhvr.handle_timeout();
-    else
-      seq.erase_and_advance();
+    } else {
+      seq.erase(post_condition);
```

Comment:
Calling `erase(false)` does nothing in both implementations. Isn't a more straightforward fix `if (!post_condition) seq.erase_and_advance();`? That wouldn't require any changes outside of `receive_impl`. Am I missing something or are you overcomplicating the solution?